osu.Game/Skinning/SkinManager.cs

```diff
@@ -104,7 +104,7 @@ public void SelectRandomSkin()
         protected override string ComputeHash(SkinInfo item, ArchiveReader reader = null)
         {
             // we need to populate early to create a hash based off skin.ini contents
-            if (item.Name?.Contains(".osk") == true)
+            if (item.Name?.EndsWith(".osk", StringComparison.OrdinalIgnoreCase) == true)
```

Comment:
Not sure about this `Contains()` -> `EndsWith()` change presented without comment. May be fine to do this, may not be. `EndsWith()` does seem to be more correct to me but there might have been a reason for doing `Contains()`.

Reply:
Whoops forgot to note that, good point, perhaps it was done with compatibility in mind.

Reply:
Not sure if it was the intention here, but leaving as `Contains` would handle potential cases of things like `skin.osk.zip` which can potentially occur depending on how the file was downloaded (though I'm not sure it can get to this point in import with a `.zip` extension currently.